raiden/tests/unit/test_channel.py

```diff
@@ -710,18 +709,23 @@ def test_automatic_dispute(raiden_network, deposit, settle_timeout, reveal_timeo
     chain0 = app0.raiden.chain
     wait_until_block(chain0, chain0.block_number() + 1)
 
+    assert channel0.external_state.closed_block != 0
+    assert channel1.external_state.closed_block != 0
+
     # wait until the settle timeout has passed
     settle_expiration = chain0.block_number() + settle_timeout
     wait_until_block(chain0, settle_expiration)
-    # manually call settle (automatic settling does not seem to work)
-    # TODO: ^ Find out why
-    channel1.external_state.settle()
-    gevent.sleep(1)
+
+    # the settle event must be set
+    assert channel0.settle_event.wait(timeout=60)
+    assert channel1.settle_event.wait(timeout=60)
```

Comment:
@LefterisJP re: https://github.com/raiden-network/raiden/pull/391#discussion_r101255574

This passes now, but I had to use a timeout well in the range of 3 blocks to make it work. I expect the solution to be somewhat brittle :(